From d558e69c04c354105863c325a88647207489e6f1 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Mon, 23 Nov 2020 14:22:52 +0200
Subject: [PATCH] [SW-25120] gaudi: sp sram: write to init ecc

In order to init the ECC in SP SRAM, write initial value to it before
using it.

Change-Id: Ia3ee743ec3f310dcf4a6b7c381bec3d15122cd43
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 7cfad43..f4a6169 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -324,3 +324,13 @@
 	memcpy(hl_ts_data, gaudi_hl_ts_data,
 		sizeof(struct hl_ts_cfg) * HL_ASIC_TS_MAX_COUNT);
 }
+
+void preboot_sp_sram_init(void)
+{
+	register int i;
+	u32 *data = (u32 *)(HL_SP_SRAM_BASE);
+
+	/* SP SRAM access is limited to 32bit */
+	for (i = 0; i < HL_SP_SRAM_SIZE / sizeof(u32); i++)
+		data[i] = 0x5a5a5a5a;
+}
diff --git a/gaudi/common.h b/gaudi/common.h
index 275300c..da798f5 100644
--- a/gaudi/common.h
+++ b/gaudi/common.h
@@ -11,6 +11,9 @@
 
 #define SRAM_SIZE			0x1400000	/* 20MB */
 
+#define HL_SP_SRAM_BASE			0x7FFBFE0000
+#define HL_SP_SRAM_SIZE			0x10000
+
 #define HL_FW_BOOT_PTRS_BASE		0x60000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xA0000
 
diff --git a/preboot/preboot.c b/preboot/preboot.c
index 577a264..e3a6e3b 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -203,6 +203,7 @@
 __weak void preboot_security_cleanup_before_run(void) {}
 __weak void preboot_led_on(void) {}
 __weak void sram_scrambling_init(void) {}
+__weak void preboot_sp_sram_init(void) {}
 
 static bool preboot_is_overlap_address_range(u64 addr, int len)
 {
@@ -243,6 +244,8 @@
 
 	hl_uart_init();
 
+	preboot_sp_sram_init();
+
 	preboot_security_init();
 	preboot_security_dynamic_init();
 	cpu_boot_sts_set(CPU_BOOT_STATUS_SECURITY_READY);
